geth/peers/syncmode_test.go

```diff
@@ -0,0 +1,88 @@
+package peers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type TopicPoolSyncModeSuite struct {
+	suite.Suite
+
+	fastMode      time.Duration
+	slowMode      time.Duration
+	fastModeLimit time.Duration
+
+	sync   *syncStrategy
+	period <-chan time.Duration
+}
+
+func TestTopicPoolSyncModeSuite(t *testing.T) {
+	suite.Run(t, new(TopicPoolSyncModeSuite))
+}
+
+func (s *TopicPoolSyncModeSuite) SetupTest() {
+	s.fastMode = time.Millisecond
+	s.slowMode = time.Millisecond * 100
+	s.fastModeLimit = time.Millisecond * 50
+	s.sync = newSyncStrategy(s.fastMode, s.slowMode, s.fastModeLimit)
+	s.period = s.sync.Start()
+}
+
+func (s *TopicPoolSyncModeSuite) TearDown() {
+	s.sync.Stop()
+}
+
+func (s *TopicPoolSyncModeSuite) TestSyncStart() {
+	s.Equal(s.fastMode, <-s.period)
+}
+
+func (s *TopicPoolSyncModeSuite) TestSyncUpdate() {
+	s.Equal(s.fastMode, <-s.period)
+
+	// should stay with fast mode
+	s.sync.Update(0, 1, 2)
+	select {
+	case <-s.period:
+		s.FailNow("period should not be updated")
+	default:
```

Comment:
is there any reason to have an empty `default:` case?

Reply:
Yeah, so the case here is that I want to check if there is anything in `s.period` channel. If there is anything, `case <-s.prediod` will be selected and the test will fail. `default:` case will just move the execution on in case there is nothing in `s.period`, what is expected in this test.

Reply:
No, this will block just like `<-s.period` without `select`.